internal/cloud/gcp.go

```diff
@@ -116,11 +116,12 @@ func (gcp *GCP) MountBucket(podMetadata *metav1.ObjectMeta, podSpec *corev1.PodS
 	return fmt.Errorf("container not found: %s", req.Container)
 }
 
-func (gcp *GCP) AssociateServiceAccount(sa *corev1.ServiceAccount) {
-	if sa.Annotations == nil {
-		sa.Annotations = map[string]string{}
-	}
-	sa.Annotations["iam.gke.io/gcp-service-account"] = fmt.Sprintf("substratus-%s@%s.iam.gserviceaccount.com", sa.Name, gcp.ProjectID)
+func (gcp *GCP) GetPrincipal(sa *corev1.ServiceAccount) string {
+	return fmt.Sprintf("substratus@%s.iam.gserviceaccount.com", gcp.ProjectID)
+}
+
+func (gcp *GCP) AssociatePrincipal(sa *corev1.ServiceAccount) {
+	sa.Annotations["iam.gke.io/gcp-service-account"] = gcp.GetPrincipal(sa)
```

Comment:
Hmm I don't love that we'd have provider specifics for annotations here. Potentially fine for now but I wonder what other options we have.

Reply:
I think the only way to solve that would through K8s OSS to standardize on how workload identity should work

Reply:
Derp, this is a part of the Cloud interface - we're already doing exactly what I was suggesting here! Never mind me!